collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java

```diff
@@ -170,6 +174,18 @@ public void onCreate(Bundle savedInstanceState) {
         }
     }
 
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        outState.putSerializable(REPEAT_GROUP_PICKER_INDEX_KEY, repeatGroupPickerIndex);
+        super.onSaveInstanceState(outState);
+    }
+
+    protected void restoreInstanceState(Bundle state) {
+        if (state == null) {
```

Comment:
Why not just 
```
if (state != null) {
    repeatGroupPickerIndex = (FormIndex) state.getSerializable(REPEAT_GROUP_PICKER_INDEX_KEY);
}
```
then you don't need that return.
Or even if you created a separate method you can call it only if `state != null`

Reply:
I personally like to check edge cases first and perform default logic at the end rather than nesting default logic inside conditions. I find that often leads to more maintainable code. No strong preference, it's just habit. I'm happy to change it if you want.

Reply:
Generally, I wouldn't ask you to fix it but usually, in the whole project, we follow my approach. We don't have any check code rule for that unfortunately so I'm trying to be like a static code analyzer :smile:

Reply:
Gotcha, I'll try to remember! Updated.